pandas/_libs/writers.pyx

```diff
@@ -90,8 +85,9 @@ def convert_json_to_lines(object arr):
     cdef:
         Py_ssize_t i = 0, num_open_brackets_seen = 0, length
         bint in_quotes = 0, is_escaping = 0
-        ndarray[uint8_t] narr
-        unsigned char v, comma, left_bracket, right_brack, newline
+        ndarray[uint8_t, ndim=1] narr
+        unsigned char v, newline, comma, left_bracket, right_bracket, quote
```

Comment:
LGTM.  Only nitpick while we're here: can you change `v` to `val` just to avoid 1-character names